File: SurgSign/Sources/SurgSign/Models/Priority.swift
import SwiftUI

/// Clinical priority level for a patient, ordered from most to least urgent.
enum Priority: String, Codable, CaseIterable, Comparable {
    case high
    case medium
    case low
    case stable

    // MARK: - Sort Order

    /// Numeric sort order where lower values represent higher priority.
    var sortOrder: Int {
        switch self {
        case .high:   return 0
        case .medium: return 1
        case .low:    return 2
        case .stable: return 3
        }
    }

    // MARK: - Display

    /// Theme color associated with this priority level.
    var color: Color {
        switch self {
        case .high:   return .surgRed
        case .medium: return .surgAM
        case .low:    return .surgGreen
        case .stable: return .gray
        }
    }

    /// Human-readable label (capitalized).
    var label: String {
        rawValue.capitalized
    }

    /// SF Symbol name for visual indicators.
    var icon: String {
        switch self {
        case .high:   return "exclamationmark.triangle.fill"
        case .medium: return "arrow.up.circle.fill"
        case .low:    return "arrow.down.circle.fill"
        case .stable: return "checkmark.circle.fill"
        }
    }

    // MARK: - Comparable

    static func < (lhs: Priority, rhs: Priority) -> Bool {
        lhs.sortOrder < rhs.sortOrder
    }
}

File: SurgSignDesktop/SurgSignDesktop/Models/Priority.cs
using System;

namespace SurgSignDesktop.Models;

/// <summary>
/// Clinical priority level for a patient, ordered from most to least urgent.
/// Raw string values match the Swift <c>Priority</c> enum so the QR wire format stays identical.
/// </summary>
public enum Priority
{
    High,
    Medium,
    Low,
    Stable
}

public static class PriorityExtensions
{
    /// <summary>
    /// Returns the raw Swift-compatible string for this priority.
    /// </summary>
    public static string RawValue(this Priority priority) => priority switch
    {
        Priority.High => "high",
        Priority.Medium => "medium",
        Priority.Low => "low",
        Priority.Stable => "stable",
        _ => "stable"
    };

    /// <summary>
    /// Capitalized display label.
    /// </summary>
    public static string Label(this Priority priority) => priority switch
    {
        Priority.High => "High",
        Priority.Medium => "Medium",
        Priority.Low => "Low",
        Priority.Stable => "Stable",
        _ => "Stable"
    };

    /// <summary>
    /// Sort order — lower is higher priority.
    /// </summary>
    public static int SortOrder(this Priority priority) => priority switch
    {
        Priority.High => 0,
        Priority.Medium => 1,
        Priority.Low => 2,
        Priority.Stable => 3,
        _ => 3
    };

    /// <summary>
    /// Parse a Swift raw value back into a <see cref="Priority"/> enum.
    /// Unknown values fall back to <see cref="Priority.Stable"/>.
    /// </summary>
    public static Priority FromRawValue(string? raw) => raw switch
    {
        "high" => Priority.High,
        "medium" => Priority.Medium,
        "low" => Priority.Low,
        "stable" => Priority.Stable,
        _ => Priority.Stable
    };
}
